include/heuristic/HeuristicMapper.hpp

```diff
@@ -341,16 +452,23 @@ inline bool operator<(const HeuristicMapper::Node& x,
 
 inline bool operator>(const HeuristicMapper::Node& x,
                       const HeuristicMapper::Node& y) {
+  // order nodes by costFixed + costHeur + lookaheadPenalty (increasing)
+  // then by validMapping (true before false)
+  // then by costHeur + lookaheadPenalty (increasing),
+  //          equivalent to ordering by costFixed (decreasing)
+  // then by the amount of validly mapped 2q gates (decreasing)
+  // then by the qubit mapping (lexicographically) as an arbitrary but
+  //          consistent tie-breaker
   const auto xcost = x.getTotalCost();
   const auto ycost = y.getTotalCost();
   if (std::abs(xcost - ycost) > 1e-6) {
     return xcost > ycost;
   }
 
-  if (x.done) {
+  if (x.validMapping && !y.validMapping) {
     return false;
   }
-  if (y.done) {
+  if (y.validMapping && !x.validMapping) {
     return true;
   }
 
```

Comment:
Technically this could be simplified to something like 
```c++
if (x.validMapping != y.validMapping) {
  return y.validMapping;
}
```

Reply:
That's true, thank you!